dScripts/02_server/Map/AM/AmTeapotServer.cpp

```diff
@@ -1,14 +1,21 @@
 #include "AmTeapotServer.h"
 #include "InventoryComponent.h"
 #include "GameMessages.h"
-
+#include "Item.h"
 
 void AmTeapotServer::OnUse(Entity* self, Entity* user) {
 	auto* inventoryComponent = user->GetComponent<InventoryComponent>();
 	if (!inventoryComponent) return;
 
-	if (inventoryComponent->GetLotCount(BLUE_FLOWER_LEAVES) >= 10) {
-		inventoryComponent->RemoveItem(BLUE_FLOWER_LEAVES, 10);
+	auto* blueFlowerItem = inventoryComponent->FindItemByLot(BLUE_FLOWER_LEAVES, eInventoryType::ITEMS);
```

Comment:
I believe this will always return the smallest stack.

Reply:
I believe best practice is to not interact with the items themselves and let the inventory component / inventory class handle stuff.

Reply:
The problem is its not looking in the correct spot.  Remove item only looks in the expected inventory of the lot and not in other places said lot may be.  What is the proposed solution for this?

Reply:
Should there be a check where if there is some count X left, check the mirrored vault inventory for the remaining items?

Reply:
Not sure there's currently a neat way of doing this. One easy solution would be to add an option or another method to the inventory system to look for the biggest stack, alternatively adding even more functionality to the inventory system to be able to handle such a case.